### PR TITLE
[SEDONA-340] Add RS_ConvexHull | [SEDONA-341] Move RS_Envelope to Geometry Functions

### PR DESCRIPTION
**Pull request by iGN5117**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-340 and https://issues.apache.org/jira/browse/SEDONA-341. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
- Add RS_ConvexHull
- Move RS_Envelope to Geometry Functions

## How was this patch tested?
- Added new tests
- Verified existing tests
- Compiled documentation

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/apache/sedona/blob/master/pom.xml#L29) in since `vX.Y.Z` format.

### CODE REVIEW

File: sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterAccessors.scala [hunk 1]
```diff
@@ -18,12 +18,12 @@
...
 package org.apache.spark.sql.sedona_sql.expressions.raster
 
-import org.apache.sedona.common.raster.RasterAccessors
+import org.apache.sedona.common.raster.{GeometryFunctions, RasterAccessors}
 import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.sedona_sql.expressions.InferrableFunctionConverter._
 import org.apache.spark.sql.sedona_sql.expressions.InferredExpression
 
-case class RS_Envelope(inputExpressions: Seq[Expression]) extends InferredExpression(RasterAccessors.envelope _) {
+case class RS_Envelope(inputExpressions: Seq[Expression]) extends InferredExpression(GeometryFunctions.envelope _) {
```
Comment (jiayuasu) [inline sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterAccessors.scala:26]: Please move this to GeometryFunction.scala as well.

File: common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        GridCoverage2D emptyRaster = RasterConstructors.makeEmptyRaster(1, 5, 10, 156, -132, 5, 10, 3, 5, 0);
+        Geometry convexHull = GeometryFunctions.convexHull(emptyRaster);
+        String expected = "POLYGON ((156 -132, 181 -107, 211 -7, 186 -32, 156 -132))";
+        String actual = Functions.asWKT(convexHull);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testConvexHullFromRasterFile() throws FactoryException, TransformException, IOException {
+        GridCoverage2D raster = rasterFromGeoTiff(resourceFolder + "raster/test1.tiff");//RasterConstructors.makeEmptyRaster(1, 5, 10, 156, -132, 5, 10, 3, 5, 0);
```
Comment (jiayuasu) [inline common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java:56]: Remove the comment. Thanks.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java (hunk 2 above)
Comment (iGN5117) [reply common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java]: Done

File: sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterAccessors.scala (hunk 1 above)
Comment (iGN5117) [reply sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterAccessors.scala]: Moved

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**